README.md

```diff
@@ -1,207 +1,249 @@
+| ---------- | --------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --- |
+| shape      | [integer] | **REQUIRED.** Shape of the input n-dimensional array ($N \times C \times H \times W$), including the batch size dimension. The batch size dimension must either be greater than 0 or -1 to indicate an unspecified batch dimension size.                                                                                                       |     |
+| dim_order  | string    | **REQUIRED.** How the above dimensions are ordered within the `shape`. `bhw`, `bchw`, `bthw`, `btchw` are valid orderings where `b`=batch, `c`=channel, `t`=time, `h`=height, w=width.                                                                                                                                                         |     |
+| data_type  | enum      | **REQUIRED.** The data type of values in the n-dimensional array. For model inputs, this should be the data type of the processed input supplied to the model inference function, not the data type of the source bands. Use one of the [common metadata data types](https://github.com/stac-extensions/raster?tab=readme-ov-file#data-types). |     |
+
+Note: It is common in the machine learning, computer vision, and remote sensing communities to refer to rasters that are inputs to a model as arrays or tensors. Array Objects are distinct from the JSON array type used to represent lists of values.
+
+### Model Output Object
+
+| Field Name               | Type                                          | Description                                                                                                                                                                                            |
+| ------------------------ | --------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| task                     | [Task Enum](#task-enum)                       | **REQUIRED.** Specifies the Machine Learning task for which the output can be used for.                                                                                                                |
+| result_array             | [[Result Array Object](#result-array-object)] | The list of output arrays/tensors from the model.                                                                                                                                                      |
```

Comment:
> Since this definition is itself inside an array (multiple outputs), and defines under it its shape as an array, this array type [Result Array Object] feels redundant. Is it simply a typo?

No, multiple distinct n-dim arrays can be returned from a single model, so I think we need to capture that. A common case is foundational models that return a dictionary of feature maps, each feature map key being a separate n-dimensional array.

> In contrast, the "Array Object" used under "Model Input Object" is defined as a single object, which makes sense in my opinion.

yeah Model Input Object is analogous to Result Array Object. There can be multiple of each. I'm down to change the names to align them, "Result Object" could work.


> In order words, input_array does not contain a JSON-array, but a JSON-object.
> I believe that will cause a lot of confusion for users.

Coming from an ML engineering background with not a lot of JSON and Javascript terminology experience, I do think "Array" is going to be the most interpretable word for scientists and ml practitioners. The model inputs are widely understood as n-dim arrays and using the word "array" will help guide them on understanding what info to put in these metadata objects. I think if we call these out as "Array Object" then developers will understand it's an object and not a JSON/Javascript array.

But if you think this would confuse a large subset of folks who work a lot in Javascript and JSON, I'm ok with removing "array" from object names. 

But I do think we need to keep the use of "array" in object fields. "array" has a clear meaning across ml practitioner communities across languages that can apply to a lot of different kinds of sensors and other data inputs. replacing this with "tensor" (too deep learning niche) or "matrix" (not widely used in ml practitioner communities) or "image" (too specific to optical) would imo be more confusing. I think ml practitioners more so than Javascript developers are going to be filling out these metadata fields and it is important to meet them where their vocabulary is at.

Reply:
I think the term "Object" is also problematic, but I try to stick to the usual STAC-way of describing the fields, which typically uses the (JSON) "Object" term, since the specification describes a JSON-schema.

What would be the best is to use something like "Input Definition" and "Result Definition". Those "Definitions" then contain `shape` and `data_type`, which are very self-explanatory for ML practitioners. 

I'll give it a try refactoring a few names, and we can see after if it needs more adjustments.

Reply:
Maybe "Input Structure" and "Result Structure" ?

Reply:
Added to https://github.com/rbavery/dlm-extension/pull/2